File: src/main/java/com/rh360/rh360/service/FinanceSummaryService.java
package com.rh360.rh360.service;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.YearMonth;
import java.util.UUID;

import org.springframework.stereotype.Service;

import com.rh360.rh360.dto.FinanceSummaryResponse;

@Service
public class FinanceSummaryService {

    private final SalaryService salaryService;
    private final ExpenseService expenseService;

    public FinanceSummaryService(SalaryService salaryService, ExpenseService expenseService) {
        this.salaryService = salaryService;
        this.expenseService = expenseService;
    }

    public FinanceSummaryResponse summaryForMonth(UUID userId, String referenceMonth) {
        YearMonth ym = YearMonth.parse(referenceMonth);
        LocalDate from = ym.atDay(1);
        LocalDate to = ym.atEndOfMonth();

        BigDecimal income = salaryService.getIncomeForMonth(userId, referenceMonth);
        BigDecimal expenses = expenseService.getExpensesForMonth(userId, from, to);

        FinanceSummaryResponse resp = new FinanceSummaryResponse();
        resp.setReferenceMonth(referenceMonth);
        resp.setTotalIncome(income);
        resp.setTotalExpenses(expenses);
        resp.setBalance(income.subtract(expenses));
        return resp;
    }
}

File: src/finance-summary/finance-summary.service.ts
import { Injectable } from '@nestjs/common';
import { FinanceExpensesService } from '../finance-expenses/finance-expenses.service';
import { FinanceSalariesService } from '../finance-salaries/finance-salaries.service';
import { FinanceSummaryResponse } from './dto/finance-summary-response.dto';

@Injectable()
export class FinanceSummaryService {
  constructor(
    private readonly salaryService: FinanceSalariesService,
    private readonly expenseService: FinanceExpensesService,
  ) {}

  async summaryForMonth(
    userId: string,
    referenceMonth: string,
  ): Promise<FinanceSummaryResponse> {
    // Parse referenceMonth (YYYY-MM) to get first and last day
    const [year, month] = referenceMonth.split('-').map(Number);
    const from = new Date(year, month - 1, 1);
    const to = new Date(year, month, 0); // Last day of month

    const income = await this.salaryService.getIncomeForMonth(userId, referenceMonth);
    const expenses = await this.expenseService.getExpensesForMonth(userId, from, to);

    return {
      referenceMonth,
      totalIncome: income,
      totalExpenses: expenses,
      balance: income - expenses,
    };
  }
}
